dgp/utils/pose.py

```diff
@@ -0,0 +1,214 @@
+"""General-purpose class for rigid-body transformations.
+"""
+import numpy as np
+from pyquaternion import Quaternion
```

Comment:
We might want to switch to scipy quaternions at some point:
https://docs.scipy.org/doc/scipy/reference/generated/scipy.spatial.transform.Rotation.html